substratevm/src/com.oracle.svm.truffle/src/com/oracle/svm/truffle/TruffleBaseFeature.java

```diff
@@ -625,6 +628,62 @@ private static void registerReflectionAccessesForRuntimeValidation(Class<?> stor
     }
 }
 
+@TargetClass(className = "com.oracle.truffle.api.staticobject.StaticProperty", onlyWith = TruffleBaseFeature.IsEnabled.class)
+final class Target_com_oracle_truffle_api_staticobject_StaticProperty {
+    public static final class OffsetTransformer implements RecomputeFieldValue.CustomFieldValueTransformer {
+
+        private static final int svmArrayBaseOffset;
+        private static final int svmArrayIndexScale;
+
+        // We have to use reflection to access private members instead of aliasing them
+        // in the substitution class since substitutions are present only in runtime
+        private static final Method staticPropertyKindGetInternalKind;
+        private static final byte objectStaticPropertyKindByte;
+
+        static {
```

Comment:
This is ugly, but I guess the "right" way is to extract the API to the Graal SDK and use it in the `com.oracle.truffle.api`? That would be out of the scope of this PR.

Reply:
As we agreed, one reflection call to an internal method is fine, so I have re-structured the code accordingly.